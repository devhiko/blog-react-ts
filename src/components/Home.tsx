import "../styles/Blogs.style.css";
import useFetch from "../useFetch";
import BlogList from "./BlogList";

const Home = () => {
  const { data: blogs } = useFetch("http://localhost:8000/blogs");
  console.log(blogs);
  return (
    <div className="blogs">
      <h3>All Blogs</h3>
      <BlogList blogs={blogs} />
    </div>
  );
};

export default Home;
