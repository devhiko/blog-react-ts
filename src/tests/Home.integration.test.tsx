import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Home from "../components/Home";

describe("Home", () => {
  const MockHome = () => {
    return (
      <Router>
        <Home />
      </Router>
    );
  };

  beforeEach(() => {
    console.log("Test Started");
    jest.mock("../mocks/fetch");
    console.log("Test Finished");
  });

  it("should render Home page and blogs in it", async () => {
    const { findByRole, findAllByTitle, findByTitle, findAllByRole } = render(<MockHome />);
    const bloglist = await findByTitle("bloglist");
    const blogs = await findAllByTitle("blog");
    expect(bloglist).toHaveClass("blog-list");
    expect(blogs).toHaveLength(blogs.length);
    expect(blogs[0]).toHaveClass("blog");
    const checkLinks = (ids: number[]) => {
      ids.forEach(async () => {
        const link = await findByRole("link", { name: `Blog Title ${ids[0]}` });
        expect(link).toHaveAttribute("href", `/blogs/${ids[0]}`);
      });
    };
    const ids = [1, 2, 3, 4, 5];
    checkLinks(ids);
    const links = await findAllByRole("link");
    expect(links).toHaveLength(ids.length);
    expect(blogs).toMatchSnapshot();
  });
});
