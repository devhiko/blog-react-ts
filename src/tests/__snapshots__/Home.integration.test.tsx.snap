// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render Home page and blogs in it 1`] = `
Array [
  <div
    class="blog"
    title="blog"
  >
    <a
      href="/blogs/1"
    >
      <b>
        Blog Title 1
      </b>
    </a>
    <p>
      Blog Content 1
    </p>
    <small>
      <em>
        <p>
          Author: 
          Blog Author
        </p>
      </em>
    </small>
  </div>,
  <div
    class="blog"
    title="blog"
  >
    <a
      href="/blogs/2"
    >
      <b>
        Blog Title 2
      </b>
    </a>
    <p>
      Blog Content 2
    </p>
    <small>
      <em>
        <p>
          Author: 
          Blog Author
        </p>
      </em>
    </small>
  </div>,
  <div
    class="blog"
    title="blog"
  >
    <a
      href="/blogs/3"
    >
      <b>
        Blog Title 3
      </b>
    </a>
    <p>
      Blog Content 3
    </p>
    <small>
      <em>
        <p>
          Author: 
          Blog Author
        </p>
      </em>
    </small>
  </div>,
  <div
    class="blog"
    title="blog"
  >
    <a
      href="/blogs/4"
    >
      <b>
        Blog Title 4
      </b>
    </a>
    <p>
      Blog Content 4
    </p>
    <small>
      <em>
        <p>
          Author: 
          Blog Author
        </p>
      </em>
    </small>
  </div>,
  <div
    class="blog"
    title="blog"
  >
    <a
      href="/blogs/5"
    >
      <b>
        Blog Title 5
      </b>
    </a>
    <p>
      Blog Content 5
    </p>
    <small>
      <em>
        <p>
          Author: 
          Blog Author
        </p>
      </em>
    </small>
  </div>,
]
`;
